src/litserve/server.py

```diff
@@ -390,23 +390,19 @@ async def stream_predict(request: self.request_type) -> self.response_type:
 
     @staticmethod
     def generate_client_file(port: Union[str, int] = 8000):
-        src_path = os.path.join(os.path.dirname(__file__), "python_client.py")
         dest_path = os.path.join(os.getcwd(), "client.py")
 
-        try:
-            # Read the template code
-            with open(src_path) as f:
-                client_code = f.read()
-
-            # Replace the default port number with the user-specified port number
-            client_code = client_code.replace("{{PORT}}", str(port))
+        if os.path.exists(dest_path):
```

Comment:
But is it a correct behaviour?
If I changed the port I expect that the file should be regenerated with a new, correct one.

Or maybe I am missing something?

Reply:
if you already have a `client.py` that you updated based on your server (let's say image classification). You won't want LitServe to replace the file. The generate client is supposed to be static template that people can quickly edit.